2023.1/serialization/serialization.adoc

```diff
@@ -0,0 +1,519 @@
+
+Assume this enumeration:
+[source]
+----
+enumeration DaysOfWeek [id 23, key days-of-week]
+
+  literal Monday       [id 34, key monday]
+  literal Tuesday      [id 2,  key tttt]
+  literal Wednesday    [id 55, key 12398712]
+----
+
+And this concept:
+[source]
+----
+concept OpeningTime           [id 44, key time_to_open]
+  property day: DaysOfWeek    [id 42, key day]
+  property startHour: Integer [id 22, key starthour]
+  property endHour: Integer   [id 89, key endhour]
+----
+
+[source,json]
+----
+include::property-variants.json[]
+----
+
+=== Containment variants
+[source,json]
+----
+include::containment-variants.json[]
+----
+
+*node* with *id* `cff` is outside the processed document.
+
+=== Reference variants
+We support different kinds of targets.{fn-org57}
+[source,json]
+----
+include::reference-variants.json[]
+----
+
+=== Annotation variants
+For this example, we need to define some annotations and their annotated concepts.
+
+NOTE: The format used in this example is non-normative.
+
+Assume these annotations:
+[source]
+----
+annotation Docu [id 23, key docuAnn]
+  multiple = true
+  annotates = IDocumentable
+  property docu: String [id 34, key Docu-docu]
+
+annotation ExtendedDocu extends Docu [id 20, key docuExtended]
+  property moreDocu: String [id mds, key MDS]
+
+annotation Marker [id 22, key myMarker]
+  multiple = false
+  annotates = Node
+
+annotation TrashCan [id 99, key throwAway]
+  multiple = false
+  annotates = Node
+  containment trash: 0..* Node  [id 2, key tat]
+
+interface JavaInfo [id 33, key jv]
+  property javaVersion: String  [id 33a, key jvA]
+
+annotation MappedToClass implements JavaInfo [id mtc, key MTC]
+  multiple = false
+  annotates = Classifier
+  reference javaClass: 1 BaseLanguageClass [id jjj, key JJJ]
+
+
+annotation UsesMapping implements JavaInfo [id um, key UM]
+  multiple = false
+  annotates = Feature
+  reference mapping: 1 MappedToClass [id jjj1, key JJJ1]
+----
+
+And these concepts:
+[source]
+----
+interface IDocumentable [id 50, key 51]
+
+concept BaseLanguageClass implements IDocumentable [id 60, key 61]
+
+concept OtherLanguageConcept [id 70, key otherLangConc]
+  reference usesTrashCan: 0..1 TrashCan [id 72, key usesTrashCan]
+----
+
+
+[source,json]
+----
+include::annotation-variants.json[]
+----
+
+[[versions]]
+== Versions
+
+NOTE: The term "version" is a bit ambiguous within LionWeb.{fn-org172}
+This section lists all versions as they appear in <<SerializationChunk.serializationFormatVersion>> and <<{m3}.adoc#Language.version, `Language.version`>> for languages `LionWeb_M3` and `LionWeb_builtins`.
+
+=== 2023.1
+Technical name: `2023.1`
+
+Initial version.
+Refer to <<{roadmap}.adoc#release-2023.1, roadmap>> for details.
+
+[[json-schema]]
+== JSON Schema for serialization
+[source,json]
+----
+include::serialization.schema.json[]
+----
+
+[[possible-values]]
+== Possible values for `properties`, `containments`, and `references`
+#TODO Shall this stay in?#
```

Comment:
Do we want to include this? I guess nobody ever read it.

Reply:
Doesn't matter: if it's part of the specification (or a clarification/elucidation of it), then it can stay – even just in case someone later says "[...]but you didn't properly specify this!"

Reply:
"Nobody ever read" also increases the chance it's incorrect ...

Reply:
That's absolutely true, unfortunately... Could we somehow set it apart for the next release, and not include it in the current?

Reply:
So we link to the "living" version now, but earmark it for the next release?

Reply:
> So we link to the "living" version now, but earmark it for the next release?

My bad, I mixed up the two discussion threads.

I now removed the "possible values" section from the 2023.1 document (still in the living version).